go/state/state_db_test.go

```diff
@@ -4143,6 +4144,177 @@ func TestSlotIdOrder(t *testing.T) {
 	}
 }
 
+func TestStateDB_SetTransientState_RollbackRemovesValue(t *testing.T) {
+	ctrl := gomock.NewController(t)
+	mock := NewMockState(ctrl)
+	db := CreateStateDBUsing(mock)
+
+	s := db.Snapshot()
+	db.SetTransientState(address1, key1, val1)
+	// Check that value was correctly set
+	if got, want := db.GetTransientState(address1, key1), val1; got != want {
+		t.Errorf("unexpected value, wanted %v, got %v", want, got)
+	}
+
+	db.RevertToSnapshot(s)
+	if got, want := db.GetTransientState(address1, key1), valEmpty; got != want {
+		t.Errorf("unexpected value, wanted %v, got %v", want, got)
+	}
+}
+
+func TestStateDB_SetTransientState_RollbackToPreviousValue(t *testing.T) {
+	ctrl := gomock.NewController(t)
+	mock := NewMockState(ctrl)
+	db := CreateStateDBUsing(mock)
+
+	db.SetTransientState(address1, key1, val1)
+	// Check that value was correctly set
+	if got, want := db.GetTransientState(address1, key1), val1; got != want {
+		t.Errorf("unexpected value, wanted %v, got %v", want, got)
+	}
+
+	s := db.Snapshot()
+	// Save previous value
+	want := db.GetTransientState(address1, key1)
+
+	db.SetTransientState(address1, key1, val2)
+	// Check that value was correctly set
+	if got, want := db.GetTransientState(address1, key1), val2; got != want {
+		t.Errorf("unexpected value, wanted %v, got %v", want, got)
+	}
+
+	db.RevertToSnapshot(s)
+
+	if got, want := db.GetTransientState(address1, key1), want; got != want {
+		t.Errorf("unexpected value, wanted %v, got %v", want, got)
+	}
+}
+
+func TestStateDB_TransientStorage_IsClearedAfterCallingEndTransaction(t *testing.T) {
+	ctrl := gomock.NewController(t)
+	mock := NewMockState(ctrl)
+	db := createStateDBWith(mock, defaultStoredDataCacheSize, true)
+
+	db.SetTransientState(address1, key1, val1)
+	// Check that value was correctly set
+	if got, want := db.GetTransientState(address1, key1), val1; got != want {
+		t.Errorf("unexpected value, wanted %v, got %v", want, got)
+	}
+
+	db.EndTransaction()
+	if got, want := db.GetTransientState(address1, key1), valEmpty; got != want {
+		t.Errorf("unexpected value, wanted %v, got %v", want, got)
+	}
+
+	if len(db.undo) > 0 {
+		t.Errorf("unexpected undo len, wanted: 0, got: %v", len(db.undo))
+	}
+}
+
+func TestStateDB_SetTransientState_NewestValueCanBeObtained(t *testing.T) {
+	ctrl := gomock.NewController(t)
+	mock := NewMockState(ctrl)
+	db := CreateStateDBUsing(mock)
+
+	db.SetTransientState(address1, key1, val1)
+	if got, want := db.GetTransientState(address1, key1), val1; got != want {
+		t.Errorf("unexpected value, wanted %v, got %v", want, got)
+	}
+
+	db.SetTransientState(address1, key1, val2)
+	if got, want := db.GetTransientState(address1, key1), val2; got != want {
+		t.Errorf("unexpected value, wanted %v, got %v", want, got)
+	}
+}
+
+func TestStateDB_SetTransientState_ValueIsNotSetIfSame(t *testing.T) {
```

Comment:
nit: the property referenced by the name of this case (ValueIsNotSetIfSame) is not tested in the code. You would have to check that the undo list did not grow, as it is done in the next test -- so this one is not redundant?

Reply:
You are right, this test is redundant - removed.